### PR TITLE
mkfs: do not overwrite existing manifest in s3 bucket

### LINKED ISSUE #3691: `cvmfs_server mkfs`- Fail before overwriting manifest on s3
`cvmfs_server mkfs` with a local storage backend will fail when the repository exists already, but will overrwrite the manifest on an s3 backend without warning. We should also fail here, possibly adding a `--force` option to go ahead regardless.

### PR DESCRIPTION
**Pull request by vvolkl**
When using mkfs+s3 with a bucket already containing a repository, we will just overwrite the existing manifest, but it's very unlikely that's what the user actually wants to do. For local repositories on local storage this already fails due to the existence of the repository configuration on the local disk. This patch adds a check in mkfs. If one actually does want to overwrite the manifest, it's now necessary to delete the old one manually from the bucket.

Fixes #3691

### CODE REVIEW

**Comment by vvolkl** [comment]
@cernvm-bot cloudtest

**Comment by vvolkl** [comment]
@cernvm-bot unittest

File: cvmfs/server/cvmfs_server_mkfs.sh [hunk 1]
```diff
@@ -204,6 +204,8 @@ cvmfs_server_mkfs() {
 
   # sanity checks
   check_repository_existence $name  && die "The repository $name already exists"
+  is_empty_repository_from_url $stratum0 ||
+      die "A manifest already exists at this url. Did you mean to use cvmfs_server import?"
```
Comment (jblomer) [inline cvmfs/server/cvmfs_server_mkfs.sh]: We may want to have a `force` flag or give instructions on how to really force making a new repo in the same space (manually remove the manifest & whitelist?)

**Review by jblomer — APPROVED** [review]
LGTM, one point for consideration

File: cvmfs/server/cvmfs_server_mkfs.sh (hunk 1 above)
Comment (vvolkl) [reply cvmfs/server/cvmfs_server_mkfs.sh]: Yes, I think this usecase is too rare to warrant a force flag, but I can add a note regarding instructions (just deleting the manifest should be enough)

**Pull request merged**